server/src/main/java/org/elasticsearch/cluster/routing/ExpectedShardSizeEstimator.java

```diff
@@ -31,6 +35,18 @@ public static long getExpectedShardSize(ShardRouting shardRouting, long defaultS
         );
     }
 
+    public static boolean shouldReserveSpaceForInitializingShard(ShardRouting shard, Metadata metadata) {
+        assert shard.initializing() : "Expected initializing shard, got: " + shard;
+        return switch (shard.recoverySource().getType()) {
+            case EMPTY_STORE -> false;// initializing from scratch no data to reserve space for
+            case EXISTING_STORE -> false;// initializing from prior local data. It is already accounted for in disk usage
+            case PEER -> true;
+            // partial searchable snapshot does not copy all data on initialization
+            case SNAPSHOT -> metadata.getIndexSafe(shard.index()).isPartialSearchableSnapshot() == false;
+            case LOCAL_SHARDS -> false;// shrink/split/clone uses hard-linking to copy existing shards
```

Comment:
Nit: a bit more whitespace around these comments (and some consistent placement, and maybe a few more words) would make them much easier to read IMO.